presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestReorderWindows.java

```diff
@@ -248,7 +248,7 @@ public void testNotReorderAcrossNonPartitionFilter()
                                         .specification(windowA)
                                         .addFunction(functionCall("avg", commonFrame, ImmutableList.of(QUANTITY_ALIAS))),
                                 project(
-                                        filter(RECEIPTDATE_ALIAS + " IS NOT NULL",
+                                        filter("NOT (" + RECEIPTDATE_ALIAS + " IS NULL)",
```

Comment:
btw why `IS NOT NULL` isn't the canonical form?

Reply:
Because we want the IR to be as small as possible. We already need to support NOT as a fundamental boolean operation. Note that the plan testing framework “cheats” and uses SQL expressions as a way to describe IR, so the fact that it’s the canonical IR form doesn’t mean it’s the canonical form in the SQL language.